### PR TITLE
HOSTEDCP-1542: Fixed infra-id not being defaulted first

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
- Fixes `infra-id` not being defaulted first
- Fixes platform specific validations not being executed

follow-up of https://github.com/openshift/hypershift/pull/4018 

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
If possible please let's get this in after https://github.com/openshift/hypershift/pull/4018

/hold

**Comment by stevekuznetsov** [comment]
I'm blind as a bat and my tabs didn't refresh this morning... already merged

/hold cancel

File: cmd/cluster/core/create.go
Comment (stevekuznetsov) [inline cmd/cluster/core/create.go:602]: Sorry for missing this - generally the validation expects to run before completion, could we hoist this please? I'll follow up with a Go type trick to make it impossible to make this mistake after your PR goes in.

**Review by stevekuznetsov — CHANGES_REQUESTED** [review]

File: cmd/cluster/core/create.go
Comment (muraee) [reply cmd/cluster/core/create.go:534]: the next validation at L542, checks if a HostedCluster resource with this name exist in the cluster, but If the name is invalid in the first place, we don't have to check that because `client.Get` call will fail anyway.

File: cmd/cluster/azure/create.go
Comment (stevekuznetsov) [inline cmd/cluster/azure/create.go:155]: `o.infra.Location` please!

**Comment by stevekuznetsov** [comment]
Superseded by https://github.com/openshift/hypershift/pull/4212

**Comment by muraee** [comment]
@stevekuznetsov getting this merged to unblock folks, the infra-id issue is also showing in aws.

**Pull request merged**
